### PR TITLE
👥Mb merge public-production app

### PR DESCRIPTION
**Pull request by migbro**
## Description

This is the proposed workflow for the synchronized public/production cavatica app.

Related issue https://app.zenhub.com/workspaces/d3b-bfx-5cdd720eed0dce4209e0b8a5/issues/d3b-center/bixu-tracker/746

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Pushed here https://cavatica.sbgenomics.com/u/kfdrc-harmonization/kf-reference-pipeline/apps/#kfdrc-harmonization/kf-reference-pipeline/kfdrc-alignment-workflow
- [ ] Started job here: https://cavatica.sbgenomics.com/u/kfdrc-harmonization/kf-reference-pipeline/tasks/35cbee47-3745-4dc2-983c-74911cd65f01/#, will abort since docs, not logic changed. File inputs show correctly

*NOTE* I will update File IDs once more after copying to apps-publisher project and delete old wf file.

**Test Configuration**:
* Environment: cwltool generates warnings, but considered acceptable
* Test files:

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: workflows/kfdrc_alignment_wf.cwl [hunk 1]
```diff
@@ -0,0 +1,692 @@
...
+    wxs_target_interval_list: { type: 'File?', doc: "An interval list file that contains the locations of the WXS targets (for HsMetrics)" }
+    # ADJUST TO GENERATE METRICS
+    run_hs_metrics: { type: boolean, doc: "HsMetrics will be collected. Only recommended for WXS inputs. Requires: wxs_bait_interval_list, wxs_target_interval_list" }
+    run_wgs_metrics: { type: boolean, doc: "WgsMetrics will be collected. Only recommended for WGS inputs. Requires: wgs_coverage_interval_list" }
+    run_agg_metrics: { type: boolean, doc: "AlignmentSummaryMetrics, GcBiasMetrics, InsertSizeMetrics, QualityScoreDistribution, and SequencingArtifactMetrics will be collected. Recommended for both WXS and WGS inputs." }
+    # ADVANCED
+    min_alignment_score: { type: 'int?', default: 30, doc: "For BWA MEM, Don't output alignment with score lower than INT. This option only affects output." }
+  ```
+
+  ## Input Agnostic Alignment Workflow
```
Comment (dmiller15) [inline workflows/kfdrc_alignment_wf.cwl:72]: This is the documentation I wrote for this workflow. Perhaps we should put it at the top where the WGS stuff is.

File: workflows/kfdrc_alignment_wf.cwl [hunk 2]
```diff
@@ -0,0 +1,692 @@
...
+  ![data service logo](https://github.com/d3b-center/d3b-research-workflows/raw/master/doc/kfdrc-logo-sm.png)
+
+  ## Basic Info
+  - pipeline flowchart:
+    - [draw.io](https://tinyurl.com/y952jek2)
+  - tool images: https://hub.docker.com/r/kfdrc/
+  - dockerfiles: https://github.com/d3b-center/bixtools
+  - tested with
+    - rabix-v1.0.5: https://github.com/rabix/bunny/releases/tag/v1.0.5
+    - cwltool: https://github.com/common-workflow-language/cwltool/releases/tag/1.0.20171107133715
```
[the remark this answers is not in the document]
Comment (migbro) [reply workflows/kfdrc_alignment_wf.cwl:25]: Hmm, can you let me know which release you did use, or at least approximate it? Thanks!

File: workflows/kfdrc_alignment_wf.cwl [hunk 3]
```diff
@@ -0,0 +1,692 @@
+cwlVersion: v1.0
+class: Workflow
+id: kfdrc-alignment-workflow
+label: Kids First DRC Alignment and GATK HaplotypeCaller Workflow
+doc: |
+  Kids First Data Resource Center Alignment and Haplotype Calling Workflow (bam/fastq-to-cram, gVCF optional). This pipeline follows
```
[the remark this answers is not in the document]
Comment (migbro) [reply workflows/kfdrc_alignment_wf.cwl:6]: Hmmm, why does GATK call the tool `HaplotypeCaller` then?

File: workflows/kfdrc_alignment_wf.cwl (hunk 1 above)
Comment (migbro) [reply workflows/kfdrc_alignment_wf.cwl]: Agreed, moved!

File: workflows/kfdrc_alignment_wf.cwl [hunk 4]
```diff
@@ -0,0 +1,692 @@
...
+  this can be used later on for further analysis in joint trio genotyping and subsequent refinement and deNovo variant analysis. If you would like to run this workflow using the cavatica public app, a basic primer on running public apps can be found [here](https://www.notion.so/d3b/Starting-From-Scratch-Running-Cavatica-af5ebb78c38a4f3190e32e67b4ce12bb).
+   Alternatively, if you'd like to run it locally using `cwltool`, a basic primer on that can be found [here](https://www.notion.so/d3b/Starting-From-Scratch-Running-CWLtool-b8dbbde2dc7742e4aff290b0a878344d) and combined with app-specific info from the readme below.
+   This workflow is the current production workflow, equivalent to this [git release](https://github.com/kids-first/kf-alignment-workflow/releases/tag/v2.1.1) and supersedes the [old workflow](https://github.com/kids-first/kf-alignment-workflow/tree/1.0.0a) and [public app](https://cavatica.sbgenomics.com/public/apps#kids-first-drc/kids-first-drc-alignment-workflow/kfdrc-alignment-bam2cram2gvcf/); however outputs are considered equivalent
+
+  # KFDRC Whole Genome Alignment Workflow
```
[the remark this answers is not in the document]
Comment (migbro) [reply workflows/kfdrc_alignment_wf.cwl]: Replaced with your written paragraph

File: workflows/kfdrc_alignment_wf.cwl [hunk 5]
```diff
@@ -0,0 +1,692 @@
...
+
+  ![data service logo](https://github.com/d3b-center/d3b-research-workflows/raw/master/doc/kfdrc-logo-sm.png)
+
+  ## Basic Info
+  - pipeline flowchart:
+    - [draw.io](https://tinyurl.com/y952jek2)
+  - tool images: https://hub.docker.com/r/kfdrc/
+  - dockerfiles: https://github.com/d3b-center/bixtools
+  - tested with
+    - rabix-v1.0.5: https://github.com/rabix/bunny/releases/tag/v1.0.5
```
[the remark this answers is not in the document]
Comment (migbro) [reply workflows/kfdrc_alignment_wf.cwl:39]: Hmmm, maybe we just say cavatica and point to the site after updated the cwl version?

File: workflows/kfdrc_alignment_wf.cwl (hunk 3 above)
[the remark this answers is not in the document]
Comment (dmiller15) [reply workflows/kfdrc_alignment_wf.cwl:6]: Maybe they mean it in the sense that its a "caller for haplotypes". I don't know but they're more clear in the docs: https://gatk.broadinstitute.org/hc/en-us/articles/360037225632-HaplotypeCaller

File: workflows/kfdrc_alignment_wf.cwl (hunk 2 above)
[the remark this answers is not in the document]
Comment (dmiller15) [reply workflows/kfdrc_alignment_wf.cwl:40]: Definitely want to at least say we were using cwltool 3. Looking at my PR for the public file indexing I put this: 3.0.20200324120055

File: workflows/kfdrc_alignment_wf.cwl (hunk 5 above)
[the remark this answers is not in the document]
Comment (dmiller15) [reply workflows/kfdrc_alignment_wf.cwl:39]: Yeah probably a good idea.

File: workflows/kfdrc_alignment_wf.cwl (hunk 3 above)
[the remark this answers is not in the document]
Comment (migbro) [reply workflows/kfdrc_alignment_wf.cwl:6]: I see...meh, if someone takes issue with it, we can revise.

**Pull request merged**